docs/baremetal-agent-controller.md

```diff
@@ -0,0 +1,205 @@
+Baremetal Agent Controller (a.k.a BMAC)
```

Comment:
@flaper87 Very nice.
I guess we should all get better at documentation, and this is a very good example on how to do it!!